From 8481f99b04ed0aab0d52322fd0f474dbc4ec02cf Mon Sep 17 00:00:00 2001
From: Sakari Ailus <sakari.ailus@nokia.com>
Date: Mon, 26 Jan 2009 15:19:31 -0600
Subject: [PATCH 253/426] omap34xxcam: Start ISP after sensor.

The sensor might be powered on already and could be outputting wrong
kind of data. This is actually quite probable. Avoid this by setting up
the sensor before the ISP.

Signed-off-by: Sakari Ailus <sakari.ailus@nokia.com>
---
 drivers/media/video/omap34xxcam.c |   19 +++++++++----------
 1 files changed, 9 insertions(+), 10 deletions(-)

diff --git a/drivers/media/video/omap34xxcam.c b/drivers/media/video/omap34xxcam.c
index 0f915ca..03ae2f2 100644
--- a/drivers/media/video/omap34xxcam.c
+++ b/drivers/media/video/omap34xxcam.c
@@ -780,22 +780,21 @@ static int vidioc_streamon(struct file *file, void *fh, enum v4l2_buf_type i)
 		goto out;
 	}
 
-	isp_buf_init();
-	rval = videobuf_streamon(&ofh->vbq);
+	rval = omap34xxcam_slave_power_set(vdev, V4L2_POWER_ON,
+					   OMAP34XXCAM_SLAVE_POWER_SENSOR_LENS);
 	if (rval) {
 		dev_dbg(vdev->cam->dev, "omap34xxcam_slave_power_set failed\n");
 		goto out;
 	}
 
-
-	rval = omap34xxcam_slave_power_set(vdev, V4L2_POWER_ON,
-					   OMAP34XXCAM_SLAVE_POWER_SENSOR_LENS);
-	if (!rval) {
+	isp_buf_init();
+	rval = videobuf_streamon(&ofh->vbq);
+	if (rval)
+		omap34xxcam_slave_power_set(
+			vdev, V4L2_POWER_OFF,
+			OMAP34XXCAM_SLAVE_POWER_SENSOR_LENS);
+	else
 		vdev->streaming = file;
-	} else {
-		isp_stop();
-		videobuf_streamoff(&ofh->vbq);
-	}
 
 out:
 	mutex_unlock(&vdev->mutex);
-- 
1.6.2.4

